estimagic/optimization/optimize.py

```diff
@@ -95,29 +92,111 @@ def minimize(
     criterion,
     params,
     algorithm,
-    criterion_args=None,
     criterion_kwargs=None,
     constraints=None,
     general_options=None,
     algo_options=None,
     dashboard=False,
     db_options=None,
 ):
-    """Minimize *criterion* using *algorithm* subject to *pc* and bounds.
+    """Minimize *criterion* using *algorithm* subject to *constraints* and bounds.
+    Run several optimizations if called by lists of inputs.
 
     Args:
-        criterion (function):
-            Python function that takes a pandas DataFrame with parameters as the first
+        criterion (function or list of functions):
+            Python function that takes a pandas Series with parameters as the first
             argument and returns a scalar floating point value.
 
-        params (pd.DataFrame):
+        params (pd.DataFrame or list of pd.DataFrames):
             See :ref:`params`.
 
-        algorithm (str):
+        algorithm (str or list of strings):
             specifies the optimization algorithm. See :ref:`list_of_algorithms`.
 
-        criterion_args (list or tuple):
-            additional positional arguments for criterion
+        criterion_kwargs (dict or list of dicts):
+            additional keyword arguments for criterion
+
+        constraints (list or list of lists):
+            list with constraint dictionaries. See for details.
+
+        general_options (dict):
+            additional configurations for the optimization
+
+        algo_options (dict or list of dicts):
+            algorithm specific configurations for the optimization
+
+        dashboard (bool):
+            whether to create and show a dashboard
+
+        db_options (dict):
+            dictionary with kwargs to be supplied to the run_server function.
+
+    """
+
+    arguments = process_optimization_arguments(
+        criterion=criterion,
+        params=params,
+        algorithm=algorithm,
+        criterion_kwargs=criterion_kwargs,
+        constraints=constraints,
+        general_options=general_options,
+        algo_options=algo_options,
+        dashboard=dashboard,
+        db_options=db_options,
+    )
+
+    # Find out number of optimizations
+    n_opts = (
```

Comment:
it would be nicer if process_optimization_arguments already returns this information, i.e.:

`arguments, n_opts = process_optimization_arguments(...)`

Otherwise we have very mixed levels of abstraction here.